src/Twig/Extension/CanonicalizeExtension.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Twig\Extension;
+
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class CanonicalizeExtension extends AbstractExtension
+{
+    // @todo: there are more locales which are not supported by moment and they need to be translated/normalized/canonicalized here
+    public const MOMENT_UNSUPPORTED_LOCALES = [
+        'de' => ['de', 'de-at'],
+        'es' => ['es', 'es-do'],
+        'nl' => ['nl', 'nl-be'],
+        'fr' => ['fr', 'fr-ca', 'fr-ch'],
+    ];
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(
+        TranslatorInterface $translator
+    ) {
+        $this->translator = $translator;
+    }
+
+    /**
+     * @return TwigFunction[]
+     */
+    public function getFunctions()
```

Comment:
```suggestion
    public function getFunctions(): array
```
the same of the other extensions